nixos/modules/services/web-apps/pixelfed.nix

```diff
@@ -0,0 +1,366 @@
+    DB_PASSWORD=${cfg.database.password}
+    '';
+  };
+
+  # shell script for local administration
+  artisan = pkgs.writeScriptBin "pixelfed" ''
+    #! ${pkgs.runtimeShell}
+    cd ${pixelfed}
+    sudo=exec
+    if [[ "$USER" != ${user} ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u ${user}'
+    fi
+    $sudo ${pkgs.php}/bin/php artisan $*
+  '';
+
+
+in {
+  options.services = {
+    pixelfed = {
+      enable = mkEnableOption (lib.mdDoc "the pixelfed service");
+
+      envFile = mkOption {
+        type = types.str;
+        description = lib.mdDoc "Pixelfed .env file used to configure the application";
+        default = ''
+        ENABLE_CONFIG_CACHE=true
+        APP_NAME=Pixelfed
+        APP_ENV=production
+        APP_DEBUG=true
+
+        # Instance Configuration
+        OPEN_REGISTRATION=true
+        ENFORCE_EMAIL_VERIFICATION=false
+        PF_MAX_USERS=1000
+        OAUTH_ENABLED=false
+
+        # Media Configuration
+        PF_OPTIMIZE_IMAGES=true
+        IMAGE_QUALITY=80
+        MAX_PHOTO_SIZE=15000
+        MAX_CAPTION_LENGTH=500
+        MAX_ALBUM_LENGTH=4
+
+        # Instance URL Configuration
+        APP_URL=https://127.0.0.1
+        APP_DOMAIN=127.0.0.1
+        ADMIN_DOMAIN=127.0.0.1
+        SESSION_DOMAIN=127.0.0.1
+        TRUST_PROXIES=*
+
+
+        # Redis Configuration
+        REDIS_CLIENT=predis
+        REDIS_SCHEME=tcp
+        REDIS_HOST=127.0.0.1
+        REDIS_PASSWORD=null
+        REDIS_PORT=6379
+
+        # Laravel Configuration
+        SESSION_DRIVER=database
+        CACHE_DRIVER=redis
+        QUEUE_DRIVER=redis
+        BROADCAST_DRIVER=log
+        LOG_CHANNEL=stack
+        HORIZON_PREFIX=horizon-
+
+        # ActivityPub Configuration
+        ACTIVITY_PUB=false
+        AP_REMOTE_FOLLOW=false
+        AP_INBOX=false
+        AP_OUTBOX=false
+        AP_SHAREDINBOX=false
+
+        # Experimental Configuration
+        EXP_EMC=true
+
+        ## Mail Configuration (Post-Installer)
+        MAIL_DRIVER=log
+        MAIL_HOST=smtp.mailtrap.io
+        MAIL_PORT=2525
+        MAIL_USERNAME=null
+        MAIL_PASSWORD=null
+        MAIL_ENCRYPTION=null
+        MAIL_FROM_ADDRESS=pixelfed@example.com
+        MAIL_FROM_NAME=Pixelfed
+
+        ## S3 Configuration (Post-Installer)
+        PF_ENABLE_CLOUD=false
+        FILESYSTEM_DRIVER=local
+        FILESYSTEM_CLOUD=s3
+        #AWS_ACCESS_KEY_ID=
+        #AWS_SECRET_ACCESS_KEY=
+        #AWS_DEFAULT_REGION=
+        #AWS_BUCKET=<BucketName>
+        #AWS_URL=
+        #AWS_ENDPOINT=
+        #AWS_USE_PATH_STYLE_ENDPOINT=false
+        '';
+      };
+
+      # database config taken from zabbix.nix
+      database = {
+        type = mkOption {
+          type = types.enum [ "mysql" "pgsql" ];
+          example = "pgsql";
+          default = "mysql";
+          description = lib.mdDoc "Database engine to use.";
+        };
+
+        host = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
```

Comment:
We want to use socket auth for psql by default

Reply:
I'm not going to offer advanced database setup, Pixelfed is known to be compatible with MySQL and the other databases are basically hit or miss even if they are mentioned in the documentation.
If some contributor wants to do it, they can. I don't advise for until 1.x though.